packages/data-broker/src/__tests__/services/dhis/DhisService/testPull/testPullAnalytics.js

```diff
@@ -275,23 +305,11 @@ export const testPullAnalytics = () => {
         });
 
         it('no program', async () => {
```

Comment:
I just realized that this change might suppress some errors. Just a note to self to revert the change

Reply:
Actually does this make sense to consider as a case?

Reply:
I believe so! Example use case: get a list of programs, filter based on a condition that depends on dynamic data, then fetch using the remaining programs. The result could sometimes be an empty program list.

My conclusions so far:
* It is generally beneficial to support empty lists/maps as input
  ✔ supports a more flexible/dynamic usage of the API
  ✔enhances robustness
  ✔reduces complexity by introducing less assumptions
* On the other hand, we shouldn't overuse default parameters (`programCodes = []`) unless it is for really optional params eg optional config. In other words, support empty inputs, but the client of the API is expected to explicitly provide them.
  ✔ simpler function signatures
  ✔ no need to worry about `null` values
  ✔ we essentially assert the type of the param

Reply:
Fantastic! That makes a lot of sense.